logisland-components/logisland-services/logisland-service-influxdb/logisland-service-influxdb-client/src/main/java/com/hurence/logisland/service/influxdb/InfluxDBControllerService.java

```diff
@@ -0,0 +1,872 @@
+            .allowableValues(
+                    ConsistencyLevel.ANY.toString(),
+                    ConsistencyLevel.ONE.toString(),
+                    ConsistencyLevel.QUORUM.toString(),
+                    ConsistencyLevel.ALL.toString())
+            .required(false)
+            .defaultValue(ConsistencyLevel.ANY.toString())
+            .build();
+
+    protected static final PropertyDescriptor RETENTION_POLICY = new PropertyDescriptor.Builder()
+            .name("influxdb.retention_policy")
+            .displayName("InfluxDB retention policy.")
+            .description("Determines the name of the retention policy to use. Defaults to autogen. The defined retention" +
+                    " policy must already be defined in the InfluxDB server.")
+            .defaultValue("autogen")
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .required(false)
+            .build();
+
+    @Override
+    public List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        List<PropertyDescriptor> descriptors = new ArrayList<>();
+        descriptors.add(URL);
+        descriptors.add(USER);
+        descriptors.add(PASSWORD);
+        descriptors.add(TAGS);
+        descriptors.add(FIELDS);
+        descriptors.add(MODE);
+        descriptors.add(CONSISTENCY_LEVEL);
+        descriptors.add(RETENTION_POLICY);
+        descriptors.add(TIME_FIELD);
+        descriptors.add(BATCH_SIZE);
+        descriptors.add(FLUSH_INTERVAL);
+        return descriptors;
+    }
+
+    Map<String, Set<String>> getTags()
+    {
+        return tags;
+    }
+
+    Map<String, Set<String>> getFields()
+    {
+        return fields;
+    }
+
+    Map<String, List<Object>> getTimeFields()
+    {
+        return timeFields;
+    }
+
+    CONFIG_MODE getMode()
+    {
+        return mode;
+    }
+
+    String getDatabase()
+    {
+        return database;
+    }
+
+    ConsistencyLevel getConsistencyLevel()
+    {
+        return consistencyLevel;
+    }
+
+    String getRetentionPolicy()
+    {
+        return retentionPolicy;
+    }
+
+    @Override
+    public void init(ControllerServiceInitializationContext context) throws InitializationException {
+
+        /**
+         * Get config and establish connection to influxdb
+         */
+        super.init(context);
+
+        // Database
+        if (!context.getPropertyValue(DATABASE).isSet())
+        {
+            throw new InitializationException("Database not provided");
+        }
+        database = context.getPropertyValue(DATABASE).asString();
+
+        // Configuration mode
+        if (!context.getPropertyValue(MODE).isSet())
+        {
+            throw new InitializationException("Configuration mode not provided");
+        }
+        mode = CONFIG_MODE.fromValue(context.getPropertyValue(MODE).asString());
+
+        // Parse tags and fields according to defined configuration mode
+        switch (mode)
```

Comment:
This should be done in method :
protected Collection customValidate(ValidationContext context)
instead. For the verify part (throwing exceptions)

Reply:
Done